File: yu-ai-router-go-sdk/callback.go
package yuairoutersdk

type FuncStreamCallback struct {
	OnMessageFunc  func(chunk ChatChunk)
	OnCompleteFunc func()
	OnErrorFunc    func(err error)
}

func (f FuncStreamCallback) OnMessage(chunk ChatChunk) {
	if f.OnMessageFunc != nil {
		f.OnMessageFunc(chunk)
	}
}

func (f FuncStreamCallback) OnComplete() {
	if f.OnCompleteFunc != nil {
		f.OnCompleteFunc()
	}
}

func (f FuncStreamCallback) OnError(err error) {
	if f.OnErrorFunc != nil {
		f.OnErrorFunc(err)
	}
}

File: yu-ai-router-python-sdk/yu_ai_router_sdk/callback.py
"""SDK stream callback protocol."""

from __future__ import annotations

from typing import Protocol

from yu_ai_router_sdk.models import ChatChunk


class StreamCallback(Protocol):
    """Callback for streaming responses."""

    def on_message(self, chunk: ChatChunk) -> None:
        """Called when a stream chunk arrives."""

    def on_complete(self) -> None:
        """Called when stream completes."""

    def on_error(self, error: Exception) -> None:
        """Called when stream fails."""
